### PR TITLE
Better Valve Index Support

### PR DESCRIPTION
**Pull request by InfiniteLee**
Adds better joystick support for Index controllers (movement, rotation, scrolling).
Also changes the grab-toggle behavior to only affect Vive wands.

### CODE REVIEW

File: src/systems/userinput/bindings/vive-user.js [hunk 1]
```diff
@@ -315,21 +348,31 @@ export const viveUserBindings = addSetsToBindings({
...
     },
+    {
+      src: {
+        value: rDpadEast2
+      },
+      dest: {
+        value: rSnapRight2
+      },
+      xform: xforms.rising,
+      priority: 1
```
Comment (johnshaughnessy) [inline src/systems/userinput/bindings/vive-user.js]: I think this priority value is unnecessary

Comment (InfiniteLee) [reply src/systems/userinput/bindings/vive-user.js]: Ah yes, forgot about that one, thanks.

**Comment by johnshaughnessy** [comment]
When me come across more things that are different between index / vive wands / vive focus plus then we ought to do what Dom suggested -- namely, create a new device for each of those, change `vrGamepadMappings` so that its values are arrays-of-objects-full-of-bindings instead of just objects-full-of-bindings.

**Comment by johnshaughnessy** [comment]
Another thing I thought of after submitting that review is that the bindings that use the Index/Knuckles controller axes (or buttons) ought to be separate from the vive-user bindings, since the vive wands don't want/need those (and we'll have to separate them if ever we want those axes/buttons to do different things).

**Pull request merged**